test/e2e/elasticsearch_token_propagation_test.go

```diff
@@ -0,0 +1,278 @@
+// +build token_propagation_elasticsearch
+
+package e2e
+
+import (
+	"bytes"
+	goctx "context"
+	"crypto/tls"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"net/http/cookiejar"
+	"net/url"
+	"strings"
+	"testing"
+
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	"github.com/operator-framework/operator-sdk/pkg/test/e2eutil"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"golang.org/x/net/html"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/jaegertracing/jaeger-operator/pkg/apis"
+	"github.com/jaegertracing/jaeger-operator/pkg/apis/jaegertracing/v1"
+	esv1 "github.com/jaegertracing/jaeger-operator/pkg/storage/elasticsearch/v1"
+)
+
+type TokenPropagationTestSuite struct {
+	suite.Suite
+}
+
+func (suite *TokenPropagationTestSuite) SetupSuite() {
+	t = suite.T()
+	if !isOpenShift(t) {
+		t.Skipf("Test %s is currently supported only on OpenShift because es-operator runs only on OpenShift\n", t.Name())
+	}
+	assert.NoError(t, framework.AddToFrameworkScheme(apis.AddToScheme, &v1.JaegerList{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Jaeger",
+			APIVersion: "jaegertracing.io/v1",
+		},
+	}))
+	assert.NoError(t, framework.AddToFrameworkScheme(apis.AddToScheme, &esv1.ElasticsearchList{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Elasticsearch",
+			APIVersion: "logging.openshift.io/v1",
+		},
+	}))
+	var err error
+	ctx, err = prepare(t)
+	if err != nil {
+		if ctx != nil {
+			ctx.Cleanup()
+		}
+		require.FailNow(t, "Failed in prepare")
+	}
+	fw = framework.Global
+	namespace, _ = ctx.GetNamespace()
+	require.NotNil(t, namespace, "GetNamespace failed")
+}
+
+func (suite *TokenPropagationTestSuite) TearDownSuite() {
+	// handleSuiteTearDown()
+}
+
+func (suite *TokenPropagationTestSuite) TestTokenPropagationTest() {
+	// create jaeger custom resource
+	exampleJaeger := getESJaegerInstance()
+	err := fw.Client.Create(goctx.TODO(), exampleJaeger, &framework.CleanupOptions{TestContext: ctx, Timeout: timeout, RetryInterval: retryInterval})
+	require.NoError(t, err, "Error deploying example Jaeger")
+	// defer undeployJaegerInstance(exampleJaeger)
+
+	err = e2eutil.WaitForDeployment(t, fw.KubeClient, namespace, "token-prop-collector", 1, retryInterval, timeout)
+	require.NoError(t, err, "Error waiting for collector deployment")
+
+	err = e2eutil.WaitForDeployment(t, fw.KubeClient, namespace, "token-prop-query", 1, retryInterval, timeout)
+	require.NoError(t, err, "Error waiting for query deployment")
+
+	queryHost := "token-prop-" + namespace + ".127.0.0.1.nip.io"
+	queryServicesEndpoint := "https://" + queryHost + "/api/services"
+
+	client, err := oAuthAuthorization(queryHost, "user-test-token", "any")
+
+	require.NoError(t, err)
```

Comment:
The test is failing for me here when I use a real OCP 4.2 OpenShift cluster.  Have you been testing against minishift?  I think you probably need to use the actual route here.

Reply:
I haven't tested against OCP 4.2/minishift, I'll give it a try and fix what is failing. Thanks!